spec/lib/water_drop/schemas/config_spec.rb

```diff
@@ -934,4 +934,18 @@
       it { expect(schema.call(config)).not_to be_success }
     end
   end
+
+  context 'when we validate sasl_oauth_token_provider' do
+    context 'when sasl_oauth_token_provider is nil' do
+      before { config[:kafka][:sasl_oauth_token_provider] = nil }
+
+      it { expect(schema.call(config)).to be_success }
+    end
+
+    context 'when sasl_oauth_token_provider is an object' do
```

Comment:
Wouldn't we want to validate that it responds to `token` as described in the comment?